src/MultipartStreamBuilder.php

```diff
@@ -228,4 +228,31 @@ public function setMimetypeHelper(MimetypeHelper $mimetypeHelper)
 
         return $this;
     }
+
+    /**
+     * Gets the filename from a given path.
+     *
+     * PHP's basename() does not properly support streams or filenames beginning with a non-US-ASCII character.
+     *
+     * @author Drupal 8.2
+     *
+     * @param string $path
+     *
+     * @return string
+     */
+    private function basename($path)
```

Comment:
Ping @sagikazarmark and @dbu 

Is this function overkill or is it the way to go if we want to be independent of the locale?